tests/python_tests/test_continuous_batching.py

```diff
@@ -337,3 +339,43 @@ def test_preemption_with_multinomial_n_seq(tmp_path, dynamic_split_fuse):
     # needed kv_blocks - 16 (2 blocks per sequence (30 tokens to generated text + prompt (> 2 tokens)) * (1 + 3 + 4) seq )
     scheduler_config = get_scheduler_config({"num_kv_blocks": 8, "dynamic_split_fuse": dynamic_split_fuse, "max_num_batched_tokens": 256, "max_num_seqs": 256})
     generate_and_compare_with_reference_text(models_path, multinomial_params_n_seq.prompts, multinomial_params_n_seq.ref_texts, multinomial_params_n_seq.generation_config, scheduler_config)
+
+def get_data_by_pipeline_type(model_path: Path, pipeline_type: str):
+    device = "CPU"
+    prompt = "Prompt example is"  
+    generation_config = GenerationConfig()
+    generation_config.max_new_tokens = 10
+    generation_config.do_sample = True
+    pipe = None
+    if pipeline_type == "continuous_batching":
+        scheduler_config = SchedulerConfig()
+        pipe = ContinuousBatchingPipeline(model_path, scheduler_config, device)
+        prompt = [prompt]
+        generation_config = [generation_config]
+    elif pipeline_type == "speculative_decoding":
+        generation_config.assistant_confidence_threshold = 0.4
+        pipe = LLMPipeline(model_path, device, draft_model=draft_model(model_path))
+    elif pipeline_type == "prompt_lookup_decoding":
+        generation_config.num_assistant_tokens = 5
+        generation_config.max_ngram_size = 3
+        pipe = LLMPipeline(model_path, device, prompt_lookup=True)
+    elif "llm_pipeline":
+        pipe = LLMPipeline(model_path, device)
+    else:
+        raise RuntimeError(f"{pipeline_type} is unknown pipeline type!")
+    return pipe, prompt, generation_config
+
+@pytest.mark.parametrize("pipeline_type", ["continuous_batching", "speculative_decoding", "prompt_lookup_decoding", "llm_pipeline"])
+@pytest.mark.precommit
+def test_pipelines_generate_with_streaming(tmp_path, pipeline_type):
+    model_id : str = "facebook/opt-125m"
+    opt_model, hf_tokenizer = get_hugging_face_models(model_id)
+
+    models_path : Path = tmp_path / "t_streaming" / model_id
+    convert_models(opt_model, hf_tokenizer, models_path)
+
+    pipe, input, gen_config = get_data_by_pipeline_type(models_path, pipeline_type)
+    py_streamer = lambda x: False
+    _ = pipe.generate(input, generation_config=gen_config, streamer=py_streamer)
+
+    rmtree(models_path)
```

Comment:
probably, you need to explicitly `del pipe` to ensure that it does not help a model, which holds files on FS.